src/Schema/Table.php

```diff
@@ -835,34 +791,245 @@ public function __clone()
+        $replacedImplicitIndexes = [];
+
+        foreach ($this->implicitIndexes as $name => $implicitIndex) {
+            if (! $implicitIndex->isFullfilledBy($indexCandidate) || ! isset($this->_indexes[$name])) {
+                continue;
+            }
+
+            $replacedImplicitIndexes[] = $name;
         }
 
-        return $this->trimQuotes(strtolower($identifier));
+        if ((isset($this->_indexes[$indexName]) && ! in_array($indexName, $replacedImplicitIndexes, true)) ||
+            ($this->_primaryKeyName !== null && $indexCandidate->isPrimary())
+        ) {
+            throw SchemaException::indexAlreadyExists($indexName, $this->_name);
+        }
+
+        foreach ($replacedImplicitIndexes as $name) {
+            unset($this->_indexes[$name], $this->implicitIndexes[$name]);
+        }
+
+        if ($indexCandidate->isPrimary()) {
+            $this->_primaryKeyName = $indexName;
+        }
+
+        $this->_indexes[$indexName] = $indexCandidate;
+
+        return $this;
     }
 
-    public function setComment(?string $comment) : self
+    /**
+     * @return self
+     */
+    protected function _addUniqueConstraint(UniqueConstraint $constraint)
     {
-        // For keeping backward compatibility with MySQL in previous releases, table comments are stored as options.
-        $this->addOption('comment', $comment);
+        $mergedNames = array_merge([$this->getName()], $constraint->getColumns());
+        $name        = strlen($constraint->getName()) > 0
+            ? $constraint->getName()
+            : $this->_generateIdentifierName($mergedNames, 'fk', $this->_getMaxIdentifierLength());
+
+        $name = $this->normalizeIdentifier($name);
+
+        $this->_uniqueConstraints[$name] = $constraint;
+
+        // If there is already an index that fulfills this requirements drop the request. In the case of __construct
+        // calling this method during hydration from schema-details all the explicitly added indexes lead to duplicates.
+        // This creates computation overhead in this case, however no duplicate indexes are ever added (column based).
+        $indexName = $this->_generateIdentifierName($mergedNames, 'idx', $this->_getMaxIdentifierLength());
+
+        $indexCandidate = $this->_createIndex($constraint->getColumns(), $indexName, true, false);
+
+        foreach ($this->_indexes as $existingIndex) {
+            if ($indexCandidate->isFullfilledBy($existingIndex)) {
+                return $this;
+            }
+        }
+
+        $this->implicitIndexes[$this->normalizeIdentifier($indexName)] = $indexCandidate;
 
         return $this;
     }
 
-    public function getComment() : ?string
+    /**
+     * @return self
+     */
+    protected function _addForeignKeyConstraint(ForeignKeyConstraint $constraint)
     {
-        return $this->_options['comment'] ?? null;
+        $constraint->setLocalTable($this);
+
+        if (strlen($constraint->getName()) > 0) {
```

Comment:
```suggestion
        if ($constraint->getName() !== '') {
```

Reply:
This method was being moved to follow the logical class declaration order:
* public constants
* protected constants
* private constants
* public static attributes
* protected static attributes
* private static attributes
* public attributes
* protected attributes
* private attributes
* constructor
* static public methods
* static protected methods
* static private methods
* magic methods
* public methods
* protected methods
* private methods
No work was done here.

Reply:
This was due to consistency to a previous of my comments. Because my suggested change wasn't needed and therefore not applied, this one also can stay as it is.